restructuredtext_lint/test/test.py

```diff
@@ -129,9 +129,50 @@ def test_rst_lint_many_files(self):
             # python ../cli.py test_files/valid.rst invalid.rst
             subprocess.check_output((sys.executable, rst_lint_path, valid_rst, invalid_rst))
         output = str(e.exception.output)
-        # 'rst-lint' should exit with error code 1:
-        self.assertEqual(e.exception.returncode, 1)
+        # 'rst-lint' should exit with error code 2 as linting failed:
+        self.assertEqual(e.exception.returncode, 2)
         # There should be a least one valid .rst file:
         self.assertIn('is clean', output)
         # There should be a least one invalid rst file:
         self.assertIn('WARNING', output)
+
+    def test_report_levels(self):
+        """A document which triggers linting warnings, to test --level argument"""
+        # Expect to get either two lines (both warnings), or no lines of output
+        for level, count in ((1, 2), (2, 2), (3, 0), (4, 0)):
```

Comment:
A pet peeve of mine is automation in testing. It leads to poor stack traces and debugging as we don't know which scenario was the cause for failure. I believe we should be fine testing exclusively 1 case (e.g. `level = 3; count 0 `) instead of all 4 of these